File: task4-python/refrigerator.py
# CSCI3180 Principles of Programming Languages
#
# --- Declaration ---
#
# I declare that the assignment here submitted is original except for source
# material explicitly acknowledged. I also acknowledge that I am aware of
# University policy and regulations on honesty in academic work, and of the
# disciplinary guidelines and procedures applicable to breaches of such policy
# and regulations, as contained in the website
# http://www.cuhk.edu.hk/policy/academichonesty/
#
# Assignment 2
# Name : Ling Leong
# Student ID : 1155062557
# Email Addr : alanalan0124@yahoo.com.hk

from smart_fixture import SmartFixture


class Refrigerator(SmartFixture):
    def __init__(self, name, weight, position, battery_capacity, initial_temperature):
        SmartFixture.__init__(self, name, weight, position, battery_capacity)
        self._temperature = initial_temperature

    @property
    def temperature(self):
        return self._temperature

    @temperature.setter
    def temperature(self, new_temperature):
        self._temperature = new_temperature
        print self.name + " was set to " + "%.1f" % new_temperature + " degrees Celsius."

File: task4-java/src/assg2task4/Refrigerator.java

/*
 *  CSCI3180 Principles of Programming Languages
 *
 *  --- Declaration ---
 *
 *  I declare that the assignment here submitted is original except for source
 *  material explicitly acknowledged. I also acknowledge that I am aware of
 *  University policy and regulations on honesty in academic work, and of the
 *  disciplinary guidelines and procedures applicable to breaches of such policy
 *  and regulations, as contained in the website
 *  http://www.cuhk.edu.hk/policy/academichonesty/
 *
 *  Assignment 2
 *  Name : Ling Leong
 *  Student ID : 1155062557
 *  Email Addr : alanalan0124@yahoo.com.hk
 */

package assg2task4;

/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

/**
 *
 * @author kenchan
 */
public class Refrigerator extends SmartFixture {
    protected double temperature;
    public Refrigerator(String Name, double Weight, int Position, double Battery_capacity, double Initial_temperature){
        super(Name, Weight, Position, Battery_capacity);
        this.temperature = Initial_temperature;
    }
    public void SetTemperature(double New_temperature){
        this.temperature = New_temperature;
        System.out.println(this.name + " was set to " + New_temperature + " degrees Celsius.");
    }
}
